src/openrct2/config.c

```diff
@@ -544,6 +544,17 @@ bool config_open(const utf8string path)
 
 	free(lineBuffer);
 	SDL_RWclose(file);
+
+	// Window limit must be kept within valid range.
+	if (gConfigGeneral.window_limit < WINDOW_LIMIT_MIN) {
+		gConfigGeneral.window_limit = WINDOW_LIMIT_MIN;
+		config_save_default();
```

Comment:
I think you are overthinking this. I think it should just be removed and then this PR can be merged.

Another reason this is wrong is that this is a function for just loading a config file, nothing else - and not necessarily the default config file. So you could end up loading a completely different config file and it then saves to the default config file.